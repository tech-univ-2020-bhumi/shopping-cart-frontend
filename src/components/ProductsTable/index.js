import React from 'react';
import propTypes from 'prop-types';
import * as styles from './index.module.css';

function ProductsTable(props) {
  const { filteredProducts } = props;
  const productsJSX = [];
  if (filteredProducts !== undefined) {
    const categories = Object.keys(filteredProducts);
    categories.forEach((category) => {
      productsJSX.push(<tr><td className={styles.tableCategory} colSpan={4}>{category}</td></tr>);
      const products = filteredProducts[category];
      products.forEach((product) => {
        productsJSX.push(
          <tr>
            <td>{product.name}</td>
            <td>{product.price}</td>
            <td>{product.cart}</td>
            <td>{product.price * product.cart}</td>
          </tr>,
        );
      });
    });
  }

  return (
    <div className={styles.table}>
      <table border={1} className={styles.tableClass}>
        <tr>
          <th>
            Item
          </th>
          <th>
            Price
          </th>
          <th>
            Quantity
          </th>
          <th>
            Subtotal
          </th>

        </tr>
        <tbody>
          {productsJSX}
        </tbody>

      </table>
    </div>

  );
}
ProductsTable.propTypes = {
  filteredProducts: propTypes.objectOf(propTypes.string).isRequired,
};

export default ProductsTable;
