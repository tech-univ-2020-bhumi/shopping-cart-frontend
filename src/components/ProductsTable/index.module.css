table, th, td{
    border-collapse:collapse;
    padding: 1rem; 
 }
 .tableCategory{
     background-color: rgb(196,196,196);
     text-align: left;
 }
 .tableClass{
     box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
 }