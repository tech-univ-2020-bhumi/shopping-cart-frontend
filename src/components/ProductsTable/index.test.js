import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import ProductsTable from './index';


describe('the ProductsTable component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<ProductsTable />);

    expect(asFragment()).toMatchSnapshot();
  });
});
