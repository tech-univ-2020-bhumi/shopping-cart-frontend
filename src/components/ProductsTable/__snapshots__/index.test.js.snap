// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the ProductsTable component should render correctly 1`] = `
<DocumentFragment>
  <div>
    <table
      border="1"
    >
      <tbody>
        <tr>
          <th>
            Item
          </th>
          <th>
            Price
          </th>
          <th>
            Quantity
          </th>
          <th>
            Subtotal
          </th>
        </tr>
      </tbody>
      <tbody />
    </table>
  </div>
</DocumentFragment>
`;
